### PR TITLE
Update "Compatibility notes" in `README.md`

### PR DESCRIPTION
**Pull request by WinterSilence**
Fix notice and add example

### CODE REVIEW

**Review by IonBazan — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -69,7 +69,16 @@ Compatibility notes
 ===================
 
 To write portable code between PHP5 and PHP7, some care must be taken:
-- `\*Error` exceptions must be caught before `\Exception`;
+- `\Exception` and `\*Exception` exceptions must be caught before `\Error`, `\*Error` or `\Throwable`:
```
[the remark this answers is not in the document]
Comment (WinterSilence) [reply README.md:72]: @IonBazan `symfony/polyfill-php70` not required

**Comment by nicolas-grekas** [comment]
`Throwable` is not polyfilled because it cannot be on PHP5. The current message is correct: if you need PHP5+7 compatibility, you need to catch errors explicitly and before exceptions.

**Comment by WinterSilence** [comment]
Current notice don't contain information about required package `symfony/polyfill-php70`. Code by notice without undeclared `symfony/polyfill-php70` can't work:
~~~php
try {
   ...
} catch (\Error $e) {
  // executes in PHP>=7
} catch (\Exception $e) {
  // executes in PHP>=5
}
~~~

**Comment by nicolas-grekas** [comment]
It's true we wrote this note before we decided to bump the minimum requirement here to PHP 7.1. Of course polyfill-php70 is needed for ppl that run on php5 and want some php7 feats. I'm not sure it's that important to highlight, since not many ppl remain on PHP5 thought. But we could indeed.

**Comment by WinterSilence** [comment]
@nicolas-grekas
> need update this notice in any case. Can you add info about working with symfony/polyfill-php70?

**Comment by IonBazan** [comment]
@WinterSilence since we no longer support <1.19.0 (there is no separate branch for that), I don't think we are planning to update the Readme. Nevertheless, the current instructions **are correct** as you are supposed to catch `Error` first, before `Exception` to keep the code PHP 5 and PHP 7 compatible. I know the wording there might not be the easiest to understand, but the meaning is that the order of `catch` methods matter when using polyfills.

Your code sample should be as follow:
```php
try {
   ...
} catch (\Error $e) {
  // executes in all PHP versions
} catch (\Exception $e) {
  // executes in all PHP versions
}
```

**Comment by WinterSilence** [comment]
@IonBazan if package `polyfill-php70` not required and PHP<5.6, then `Error` not declared and your code generate error

**Pull request closed without merging**